blosc/blosc2.c

```diff
@@ -1211,8 +1211,8 @@ static int blosc_d(
   int32_t compformat = (context->header_flags & (uint8_t)0xe0) >> 5u;
   int dont_split = (context->header_flags & (uint8_t)0x10) >> 4u;
   int memcpyed = context->header_flags & (uint8_t)BLOSC_MEMCPYED;
-  int32_t chunk_nbytes = *(int32_t*)(src + BLOSC2_CHUNK_NBYTES);
-  int32_t chunk_cbytes = *(int32_t*)(src + BLOSC2_CHUNK_CBYTES);
+  int32_t chunk_nbytes = sw32_(src + BLOSC2_CHUNK_NBYTES);
+  int32_t chunk_cbytes = sw32_(src + BLOSC2_CHUNK_CBYTES);
```

Comment:
I would honestly more prefer to re-read the blosc header using `blosc_read_header`. It has more checks to it. Thoughts?

Reply:
OMG, that bug was bad.  Curiously nobody never reported about this, which is probably a testimony that there is not a lot of users (well, none :-) creating chunks in one machine and decompressing them on a big-endian box.  However, this could be changing a lot for Blosc2 because frames makes data transport much more easier.

So +1 on using a `blosc_read_header` every time one need to do that.

Reply:
It looks like there are a lot of places where reading `cbytes` or `nbytes` from a chunk, so I have used `blosc_cbuffer_sizes` instead. There are a few more places I need to get. Surprisingly this is almost all of the cases in the code where `+ BLOSC2_CHUNK_*` is used in `frame.c` or `schunk.c`  - for only getting those two values. There is a separate instance that checks blosc2_flags.

I would prefer not to expose `blosc_read_header` and `blosc_header_s` if necessary.